modules/ROOT/pages/6.3-release-notes.adoc

```diff
@@ -0,0 +1,704 @@
+
+
+=== When running on Firefox, the editor.selection.getRng() API was not returning expected values on hidden editors
+
+In contrast to other supported browsers, Firefox did not return the selected object or the selection range from iframes that are hidden by the `display: none` attribute.
+
+As a consequence, when a {productname} instance was running on Firefox, the `window.getSelection()` API for an iframe returned `null` if the iframe was hidden. This, further, resulted in the `editor.selection.getRng()` and `editor.selection.getBookmark()` APIs returning incorrect data from hidden editors when Firefox was the host browser.
+
+This has been fixed in {productname} 6.3. All browsers, including Firefox, now return the correct values even when the iframe is hidden.
+
+
+=== Dragging a noneditable element before or after another noneditable element now works correctly.
+
+Elements that had the `contenteditable="false"` attribute set could not be dragged to the left of an ancestor element.
+
+For example in the following
+
+[source,html]
+----
+<p>Text in a paragraph tag with a span of <span contenteditable="false">non-editable text</span> inside.</p>
+----
+
+the string _non-editable text_ could not be dragged to before the opening `+<p>+` tag.
+
+{productname} 6.3 can place a fake insertion point next to elements with a `contenteditable="false"` attribute when another element with a `contenteditable="false"` attribute is dragged, allowing the object to be moved before and after other such elements.
+
+
+=== The restored selection or location after a redo or undo action was not scrolled into view.
+
+Prior to the {productname} 6.3 release, executing an undo or redo function in the editor did not scroll the editor to the new insertion point or selection, should the undo or redo result in the insertion point or selection being out of the current viewport.
+
+Instead, users had to manually scroll to the insertion point or selection after issuing an undo or redo command that resulted in the insertion point or selection being out of view.
+
+As of the {productname} 6.3 release, the scroll function now repositions the insertion point to the last known undo or redo location.
+
+
+=== A newline could not be inserted after an inline element with a contenteditable="false" attribute was created.
+
+In previous versions of {productname}, a newline could not be inserted when the selection was restored from a ‘bookmark’ positioning after adding an inline element with a `contenteditable="false"` attribute.
+
+This was most commonly seen by pressing the *Return* (or *Enter*) key not moving the insertion point to a new line after a merge tag was created in the editor.
+
+For {productname} 6.3, a fix was applied to the core `SelectionOverrides` settings to ensure the user selection restores from the ‘bookmark’ positioning before and after an inline element with a `contenteditable="false"` attribute is created.
```

Comment:
Smart quotes around bookmark.

Reply:
Hmm. It seems intentional. I probably just don't know the syntax. Ignore.

Reply:
It is intentional but, as the bunny who put the quotes around the word, it’s also something of a desperation device.

I believe the word is being used here as a developer’s [term-of-art](https://dictionary.com/browse/term-of-art). And this term-of-art use, while recognisable drawing from the more general term, is _not_ the broader computer-specific meaning, nor the word’s pre-computer meaning.

So the quotes are my desperation device to denote _this word doesn’t denote quite what you likely assume it does_.

I’ve removed them. In their place is the hope that the audience is developer-centric enough to parse the word as a term-of-art.